packages/ethereum-contracts/contracts/gov/SuperfluidGovernanceBase.sol

```diff
@@ -63,12 +63,21 @@ abstract contract SuperfluidGovernanceBase is ISuperfluidGovernance
     {
         if (hostNewLogic != address(0)) {
             UUPSProxiable(address(host)).updateCode(hostNewLogic);
+            UUPSProxiable(address(hostNewLogic)).castrate();
         }
         for (uint i = 0; i < agreementClassNewLogics.length; ++i) {
             host.updateAgreementClass(ISuperAgreement(agreementClassNewLogics[i]));
+            UUPSProxiable(address(agreementClassNewLogics[i])).castrate();
         }
         if (superTokenFactoryNewLogic != address(0)) {
             host.updateSuperTokenFactory(ISuperTokenFactory(superTokenFactoryNewLogic));
+
+            // the factory logic can be updated for non-upgradable hosts too,
```

Comment:
hmm, should we stick to this case at all? Not sure anymore why it was like this in the first place, and I don't think it had practical reason

Reply:
My understanding is that it's used for testing only.
Also checking that code I wonder what the point of using a proxy for the factory is. Since it's always upgradable in the host and carries no state beyond a reference to a SuperToken logic contract - which changes with a new factory anyway - it could imo always be used without proxy. Only difference to consider then is that its address would change on updates, to be supported e.g. by the subgraph.

Reply:
The ability to update it for non-upgradable hosts is needed at the moment bcs it's the only way a factory can be attached to the host.

Reply:
> bcs it's the only way a factory can be attached to the host.

is it for a good reason or it is something we can change?

Reply:
ah, never mind, I read the code again, it's clear